test/azure/RspecTests/resource_inheritance.rb

```diff
@@ -0,0 +1,57 @@
+# encoding: utf-8
+
+$: << 'RspecTests/Generated/azure_resource_inheritance'
+
+require 'rspec'
+require 'generated/resource_inheritance'
+
+include AzureResourceInheritanceModule
+
+describe 'ResourceInheritance' do
+  before(:all) do
+    @base_url = ENV['StubServerURI']
+
+    dummyToken = 'dummy12321343423'
+    @credentials = MsRest::TokenCredentials.new(dummyToken)
+
+    @client = AzureResourceInheritanceTest.new(@credentials, @base_url)
+  end
+
+  it 'should generate expected resource models' do
+    modules = Module.const_get('AzureResourceInheritanceModule::Models')
+
+    # Should generate AzureResource class
+    azure_resource = modules.const_get('AzureResource')
+    expect(azure_resource.is_a?(Class)).to be_truthy
+
+    # Should generate LikeAzureResource class
+    like_azure_resource = modules.const_get('LikeAzureResource')
+    expect(like_azure_resource.is_a?(Class)).to be_truthy
+
+    # Should generate AzureResourceAdditionaProperties class
+    azure_resource_additional_properties = modules.const_get('AzureResourceAdditionaProperties')
+    expect(azure_resource_additional_properties.is_a?(Class)).to be_truthy
+
+    # Should not generate Resource class
+    expect { modules.const_get('Resource') }.to raise_error(NameError)
```

Comment:
"Resource" class would exist from MsRestAzure due to the declaration at https://github.com/Azure/autorest.ruby/pull/9/files#diff-f1567bd7380ca2e7818c4047e9cdd67dR194 right? 
Is this just checking that we did not _generate_ a class withe the name "Resource"?